rmq/consumer/connector.go

```diff
@@ -82,6 +82,35 @@ func New(connConfig ConnectionConfig, queueConfig QueueConfig, queueBindConfig Q
 	return nil
 }
 
+// NewConnection creates a new connection to RabbitMQ. For use when you want to initialize queues and exchanges or verify that they are present
+func NewConnection(connConfig ConnectionConfig, logger ILogger) (*amqp.Connection, error) {
+	attempts := 0
+	for attempts <= connConfig.ReconnectMaxAttempt {
+		logger.Out("RMQ-CONSUMER", "Connecting to RabbitMQ")
+		// Make a connection to RMQ
+		conn, err := amqp.Dial(connConfig.ConnURIs[0])
+		if err != nil {
+			logger.Error("ERR_RMQ-CONSUMER_FAIL-CONNECT", err)
+			time.Sleep(connConfig.ReconnectInterval)
+			// Wait before retrying
+			continue
+		}
+		logger.Out("RMQ-CONSUMER", "Connected to RabbitMQ")
+		return conn, nil
+	}
+	return nil, fmt.Errorf("failed to connect to RabbitMQ after %d attempts", connConfig.ReconnectMaxAttempt)
```

Comment:
what is the default max attempt?

Reply:
Default is set to 3. 
`// DefaultConnectionConfig returns a default connection configuration.
func DefaultConnectionConfig(connURIs []string) ConnectionConfig {
	return ConnectionConfig{
		ConnURIs:            connURIs,
		ReconnectInterval:   5 * time.Second,
		ReconnectMaxAttempt: 3,
	}
}`

Reply:
should we make that more obvious like at the top of file or in env or constants?